eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -602,11 +602,11 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: A
    *         - true if there is a filter and msg has no timestamp, or has one that matches the filter
    *         - false otherwise
    */
-  def timestampInRange(msg: RoutingMessage, origins: Set[ActorRef], gossipTimestampFilter_opt: Option[GossipTimestampFilter]): Boolean = {
+  def timestampInRange(msg: RoutingMessage, origins: Set[GossipOrigin], gossipTimestampFilter_opt: Option[GossipTimestampFilter]): Boolean = {
     // For our own gossip, we should ignore the peer's timestamp filter.
     val isOurGossip = msg match {
       case n: NodeAnnouncement if n.nodeId == nodeParams.nodeId => true
-      case _ if origins.isEmpty || origins == Set(router) => true // if gossip doesn't come from another peer, it's ours
+      case _ if origins.exists(_.isInstanceOf[LocalGossip]) => true
```

Comment:
With the suggestion below this would become:

```suggestion
      case _ if origins.contains(LocalGossip) => true
```